app/models/product.rb

```diff
@@ -1,6 +1,5 @@
 class Product < ApplicationRecord
   validates :name, :price, :on_hand_count, presence: true
-  has_many :order_items, as: :source
-
+  has_many :order_items, as: :source, dependent: :destroy
```

Comment:
same. wantable uses a "deleted" state change to pretend things were deleted but never actually does it